$OpenBSD: patch-libavcodec_libaomenc_c,v 1.1 2019/11/07 10:13:50 ajacoutot Exp $

Change libaom default to crf=32.

Current default is 256kbps, which produces inconsistent
results (too high for low-res, too low for hi-res).
Use CRF instead, which will adapt.

Index: libavcodec/libaomenc.c
--- libavcodec/libaomenc.c.orig
+++ libavcodec/libaomenc.c
@@ -575,10 +575,11 @@ static av_cold int aom_init(AVCodecContext *avctx,
         if (enccfg.rc_end_usage == AOM_CQ) {
             enccfg.rc_target_bitrate = 1000000;
         } else {
-            avctx->bit_rate = enccfg.rc_target_bitrate * 1000;
+            enccfg.rc_end_usage = AOM_Q;
+            ctx->crf = 32;
             av_log(avctx, AV_LOG_WARNING,
-                   "Neither bitrate nor constrained quality specified, using default bitrate of %dkbit/sec\n",
-                   enccfg.rc_target_bitrate);
+                   "Neither bitrate nor constrained quality specified, using default CRF of %d\n",
+                   ctx->crf);
         }
     }
 
@@ -1091,7 +1092,7 @@ static const AVOption options[] = {
 };
 
 static const AVCodecDefault defaults[] = {
-    { "b",          "256*1000" },
+    { "b",                 "0" },
     { "qmin",             "-1" },
     { "qmax",             "-1" },
     { "g",                "-1" },
